.github/workflows/validate_python.yml

```diff
@@ -0,0 +1,39 @@
+name: Validate Python Code
```

Comment:
Nitpick: Consider making this a separate [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) file and running it as part of the build(s) for PRs instead of separately. That way we don't run the full build(s) multiple times when there's invalid syntax and a user fixes it.